distributed/shuffle/_arrow.py

```diff
@@ -20,11 +20,6 @@ def check_dtype_support(meta_input: pd.DataFrame) -> None:
             raise TypeError(
                 f"p2p does not support data of type '{column.dtype}' found in column '{name}'."
             )
-        # FIXME: Serializing custom objects to PyArrow is not supported in P2P shuffling
-        if pd.api.types.is_object_dtype(column):
-            raise TypeError(
-                f"p2p does not support custom objects found in column '{name}'."
-            )
```

Comment:
@hendrikmakait Sorry I missed this in review of https://github.com/dask/distributed/pull/7425 but `is_object_dtype` is way too generic to raise here. This includes very simple string columns and we should definitely be able to shuffle dataframes with string columns. Since this is purely about failing early, I removed this check.